adt_press/nodes/a11y_nodes.py

```diff
@@ -0,0 +1,375 @@
+import json
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Any, cast
+
+from hamilton.function_modifiers import cache
+
+from adt_press.models.config import TemplateConfig
+from adt_press.utils.html import render_template
+
+
+@cache(behavior="recompute")
+def _write_package_json(work_dir: Path) -> None:
+    pkg = work_dir / "package.json"
+
+    base = {
+        "type": "module",
+        "dependencies": {
+            "@guidepup/virtual-screen-reader": "^0.32.1",
+            "jsdom": "^24.1.0",
+            "axe-core": "^4.9.0",
+            "canvas": "^2.11.2",
+        },
+    }
+
+    if pkg.exists():
+        try:
+            current = json.loads(pkg.read_text())
+        except Exception:
+            current = {}
+        deps = current.get("dependencies", {})
+        deps.update(base["dependencies"])
+        current["dependencies"] = deps
+        current.setdefault("type", "module")
+        pkg.write_text(json.dumps(current, indent=2))
+    else:
+        pkg.write_text(json.dumps(base, indent=2))
```

Comment:
This feels a bit nuts, we are manually rewriting our node dependencies? Is that right? Can we just use npm or whatever package manager to do this? (or should this dependency just always be included?)